plugins/org.eclipse.n4js/src/org/eclipse/n4js/binaries/nodejs/NodeYarnProcessBuilder.java

```diff
@@ -0,0 +1,262 @@
+	 * @param save
+	 *            instructs npm/yarn to save installed packages in package.json. Note that yarn will always do this, so
+	 *            this flag is ignored in case yarn is used.
+	 * @return configured, operating system aware process builder for "npm/yarn uninstall" command
+	 */
+	public ProcessBuilder getUninstallNpmPackageProcessBuilder(File invocationPath, String packageName, boolean save) {
+		return installUninstallCall(invocationPath, Operation.UNINSTALL, packageName, save);
+	}
+
+	private ProcessBuilder installUninstallCall(File invocationPath, Operation operation, String packageName,
+			boolean save) {
+		Objects.requireNonNull(packageName);
+
+		Binary binary = isYarnUsed(invocationPath.toPath()) ? yarnBinaryProvider.get() : npmBinaryProvider.get();
+		if (!binary.canInstallNpmPackages()) {
+			throw new IllegalStateException("(un-)installing npm packages not supported by binary: " + binary.getId());
+		}
+
+		String simpleCommand = null;
+		switch (operation) {
+		case INSTALL:
+			simpleCommand = binary.getNpmInstallCommand(packageName.trim().isEmpty());
+			break;
+		case UNINSTALL:
+			simpleCommand = binary.getNpmUninstallCommand();
+			break;
+		}
+
+		Optional<Pair<String, String>> saveOptions = binary.getNpmSaveOptions();
+		String saveOption;
+		if (saveOptions.isPresent()) {
+			saveOption = save ? saveOptions.get().getKey() : saveOptions.get().getValue();
+		} else {
+			saveOption = ""; // the fact that save/no-save is unsupported by binary is ignored
+		}
+
+		Builder<String> builder = ImmutableList.<String> builder();
+		if (isWindows()) {
+			builder.add(WIN_SHELL_COMAMNDS);
+			builder.add(escapeBinaryPath(binary.getBinaryAbsolutePath()), simpleCommand, packageName, saveOption);
+		} else {
+			builder.add(NIX_SHELL_COMAMNDS);
+			builder.add(
+					escapeBinaryPath(binary.getBinaryAbsolutePath()) + " " + simpleCommand + " " + packageName + " "
+							+ saveOption);
+		}
+
+		return create(builder.build(), binary, invocationPath, false);
+	}
+
+	/**
+	 * Tells if yarn will be used (instead of npm) for managing npm packages in the given project folder. For now, this
+	 * method returns <code>true</code> if and only if {@link #isYarnRequired(Path) yarn is required}, but in the future
+	 * this choice might be taken by the user, e.g. via a setting in the Eclipse preferences.
+	 * <p>
+	 * For now, the caller can assume that if this method returns <code>false</code>, npm will be used for managing npm
+	 * packages, as there isn't a third tool supported, at the moment.
+	 *
+	 * @param projectFolder
+	 *            path to an N4JS or plain Javascript project folder (which may or may not be contained in a yarn
+	 *            workspace) or the workspace root of a yarn workspace.
+	 * @return <code>true</code> if yarn is used for managing npm packages at the given location, <code>false</code> if
+	 *         npm is used.
+	 */
+	public boolean isYarnUsed(Path projectFolder) {
```

Comment:
please double-check all callers to make sure it is ok to invoke this relatively costly method (accessing file system, parsing package.json, etc.).

Also copy the disclaimer from the API doc of method `NodeModulesDiscoveryHelper#getYarnWorkspaceRoot()` to this method:
```
NOTE: this method is accessing the file system, so it should be deemed expensive!
```